README
NAME
    Handel - A cart/order/checkout framework with AxKit/TT/Catalyst support

SYNOPSIS
        use Handel;
        Handel->config_class('My::ConfigReader');
    
        my $config = Handel->config;
        # $config->isa('My::ConfigReader')

DESCRIPTION
    This is a generic class containing the default configuration used by
    other Handel classes.

    To learn more about what Handel is and how it works, take a look at the
    manual.

METHODS
  config_class
    Arguments: $config_class

    Gets/sets the name of the configuration class to use. The default
    configuration class is Handel::ConfigReader.

    A Handel::Exception exception will be thrown if the specified class can
    not be loaded.

  config
    Returns an instance of the specified configuration class.

SEE ALSO
    Handel::Cart, Handel::Order, Handel::Checkout

AUTHOR
        Christopher H. Laco
        CPAN ID: CLACO
        claco@chrislaco.com
        http://today.icantfocus.com/blog/


Changes
Revision history for Perl module Handel

0.99_19 Thur June 21 21:02:36 2007
    - Storage->setup now puts non-accessor keys into %$self
    - Removed incomplete L10N Lexicons
    - Cleanup coverage before 1.0 release
    - DBIx::Class 0.08 Released & Required
    - Added AutoCommit to Handel::Test
    - Tweaked failing checkout process tests

0.99_18 Wed Apr 18 19:55:09 2007
    - Added --version to handel.pl/handel.bat
    - Fixed Handel::Compat::Currency pod
    - Updated prereq to use DBIx::Class 0.07999_02
    - Added skips for DBD::SQLite 1.13 (teh devil) causing failing tests
    - Updated Perl::Critic tests

0.99_17 Wed Mar 21 20:11:24 2007
    - Renamed ConfigReaders %Defaults to %DEFAULTS
    - Removed t/style_critic_tests. ETOOMUCHWORKFORLITTLEBENEFIT
    - Updated Test::MockObject in tests to 1.07
    - Fixed destroy in Order
    - Fixed has_column check when setting item currency columns

    BREAKING API CHANGE:
    - Handel::Currency isa Data::Currency
    - H::C::format moved to as_string; now just returns the format

0.99_16 Thur Dec 21 16:21:15 2006
    - Added \%options/order_by to search/items in Cart/Order
    - Added has_column to DBIC Storage to use ResultSource->has_column
    - Added has_column to Result/DBIC Result
    - Fixed TT demo code. Maybe that will die some day.
        Seriously. Two words: Catalyst Helpers. :-)
    - Fixed issue where Compat Currency convert failed with converter returns 0
    - Changed quickstart setup script to *_handel.pl
    - Added 0700 perms to handel.pl app script
    - Storage::process_error now only dies Handel::Exception subclasses instead
    of
        all blessed objects
    - Renamed DBIC schema classes into Schema::DBIC space, like their RDBO
    cousins

0.99_15 Sun Nov 12 17:50:52 2006
    - L10N files now use constant-like keys rather than english errors
    - L10N::es_es.pm reworked by Diego Kuperman
    - L10N::fr reworked by Pierrick DINTRAT
    - Added l10n_lexicon_synced.t to ensure language lexicons are synced with
        each other via key checks
    - Removed constraint_uuid check from Checkout order/cart since it's not
        subclass friendly (RT#19813,TKP)
    - Checkout cart now throws an exception of now order can be found matching
        the search criteria
    - Converted tests to use Handel::Test
    - Cleanup tests against Perl::Critic
    - Fixed Checkout prototype mismatch warning under perl > 5.8.8
    - Added more config tests for MP1/MP2
    - Added more Schema connect tests
    - Added more Base tests
    - Added Iterator tests
    - Fixed bug in Iterator:::DBIC where first increments the resulset
    - Fixed issue in Cat Helpers that treated MyApp::Controller:: as a valid
    name
    - Added Catalyst Model setup tests
    - Added Catalyst Scaffold Helper tests
    - Added checkout_class to Order pod
    - Added Module::Starter::Handel to help kickstart Handel projects
    - Added currency_code_column to storage so DBIC inflators can set currency
        code using another column in the row
    - Removed formatting options from Currency->convert now that it always
    returns
        a currency object instead of formatted strings
    - Reworked Handel::Currency to be more object like with its parameters
    - Added Handel::Compat::Currency, inserted automatically when loading
        Handel::Compat
    - Added compat_currency.t tests
    - Converted to Module::Pluggable::Object
    - Cart restore now throws exception on unknown mode like it was supposed to
    - Only create currency converter when calling convert for the first time
    - Added handel.pl quickstart scripts for Module::Starter::Handel apps
    - Added Catalyst::Helper::Handel to create entire cat app w/ starter
    subclasses
    - Huge test coverage. 99% w00t!
    - Mostly Perl::Critic happy...except for some test labels
    - Probably last release before 1.0!

0.99_14 Sat Sep 23 19:14:22 2006
    - Added Order::save method
    - MarkOrderSaved plugin now uses Order::save method instead of setting
        the type field directly
    - Removed constraint_uuid check from Order create/reconcile as it's not
        subclass of interchangable schema friendly (RT#19813,TKP)
    - Changed id references in search to use first primary key from Storage
        (RT#19813,TKP)
    - Order creation from cart no does can('shopper') to help w/ subclassing

0.99_13 Wed Sep 20 19:01:29 2006
    - Added NEXT to prereqs for Handel::Compat
    - Moved FormValidator::Simple from recommended to prereqs
    - Added set_default_values to Storage
    - Added check_constraints to Storage
    - Added validate_data to Storage

0.99_12 Fri Sep 15 22:28:53 2006
    - Moved Locale::Currency/Format and WebServiceX modules to requirements
    - Currency new() now takes $code and $format params
    - Added name/code to Currency
    - Added converter_class to Currency
    - Added currency_code and currency_format to Storage
    - Storage::DBIC now passes currency_code/format into DBIC inflate subs for
        currency_columns
    - Added Storage::DBIC currency_code/currency_format tests
    - Finished Perl::Critic cleanup

0.99_11 Sun Sep 10 13:33:22 2006
    - Fix examples code in AddingColumns.pod
    - Fixed pod references after DBIC Storage rearrangement
    - Added item_storage_class/item_storage to Storage so item related methods
        could get item storage config from item storage without being tied to
        top level item class
    - Moved item_class/cart_class back into Handel::Base. They don't belong in
        storage
    - Moved checkout_class back into Handel::Order. That doesn't belong in
        storage either
    - Added WritingCustomStorage to Cookbook
    - Added has_column to Storage

0.99_10 Wed Sep 06 21:02:33 2006
    - Added start of Cookbook and Cookbook/AddColumns

Makefile.PL
# $Id: /local/Handel/trunk/Makefile.PL 1684 2007-06-26T03:17:42.962862Z claco  $
use ExtUtils::MakeMaker;
use 5.008001;
use strict;
use warnings FATAL => 'all';

require Test::More;
if (Test::More->VERSION < 0.60) {
    print "\a\a\a\a\a";
    print "----------------------------------------------------------\n";
    print "Test::More < 0.61 sometimes causes test failures with\n";
    print "when comparing strigified DateTime objects.\n";
    print "Please consider updating Test::More to version 0.61\n";
    print "or greater if you experience test failures.\n";
    print "----------------------------------------------------------\n";
    sleep 5;
};

#eval 'require Handel';
#if (!$@ && $Handel::VERSION <= '1.00') {
#    print "\a\a\a\a\a";
#    print "----------------------------------------------------------\n";
#    print "BREAKING API CHANGES!!!!\n\n";
#    print "Handel 1.0+ is not API compatable with Handel <= 0.32.\n";
#    print "Please read the documentation for further details.\n";
#    print "----------------------------------------------------------\n";
#    sleep 5;
#};

if ($ENV{TEST_HTTP}) {
    eval 'use Apache::Test 1.27';
    if ($@) {
        print "Apache-Test >= 1.27 is required!\n";
    } else {
        require Apache::TestMM;
        Apache::TestMM->import(qw(test clean));

        my @defines;
        eval 'use Apache::Template';
        if (!$@) {
            push @defines, 'TT2';
        };

        ## for whatever reason, use AxKit causes issues
        ## use-ing A::A::Exception doesn't
        eval 'use Apache::AxKit::Exception';
        if (!$@) {
            push @defines, 'AXKIT';
        };

        push(@ARGV, '-defines', join(' ', @defines)) if scalar @defines;
        Apache::TestMM::filter_args();
        Apache::TestMM::generate_script('t/TEST');
        Apache::TestMM::generate_script('t/SMOKE');
    };
} else {
    print 'Set TEST_HTTP to enable Apache-Test/HTTP tests', "\n";
};

my @clean = qw(
    smoke-report-*
    t/TEST
    t/SMOKE
    t/logs
    t/conf/apache_test_config.pm
    t/conf/extra.conf
    t/conf/httpd.conf
    t/conf/modperl_inc.pl
    t/conf/modperl_startup.pl
    t/htdocs/index.html
    t/var
);

WriteMakefile(
    NAME         => 'Handel',
    VERSION_FROM => 'lib/Handel.pm',
    AUTHOR       => 'Christopher H. Laco <claco@chrislaco.com>',
    ABSTRACT     => 'A cart/order/checkout framework with AxKit/TT/Catalyst support',
    PREREQ_PM    => {
        'DBIx::Class'              => '0.08002',
        'DBIx::Class::UUIDColumns' => '0.02000',
        'DBIx::Class::Validation'  => '0.02000',
        'Data::Currency'           => '0.04002',
        'Class::Accessor::Grouped' => '0.03',
        'Class::ISA'               => '0',
        'Class::Inspector'         => '0',
        'Clone'                    => '0',
        'Error'                    => '0.14',
        'Locale::Maketext'         => '1.06',
        'Module::Pluggable'        => '3.1',
        'Module::Starter'          => '1.42',
        'Scalar::Util'             => '0',
        'Tie::Hash'                => '0',
        'DateTime'                 => '0',
        'DateTime::Format::MySQL'  => '0',
        'Locale::Currency'         => '2.07',
        'Locale::Currency::Format' => '1.22',
        'FormValidator::Simple'    => '0.17',
        'Finance::Currency::Convert::WebserviceX' => '0.03',
        'NEXT'                     => '0',
        'FileHandle'               => '0',
        'File::Spec::Functions'    => '0',
        'File::Path'               => '0',
        'SQL::Translator'          => '0.08',
        'Getopt::Long'             => '0',
        'Pod::Usage'               => '0',
        'DBD::SQLite'              => '0',
    },
    EXE_FILES => [
        'script/handel.pl'
    ],
    NO_META => 1,
    dist => {
        PREOP => 'pod2text lib/Handel.pm > $(DISTVNAME)/README',
    },
    clean => { FILES => join ' ', @clean }
);
